sql/analyzer/costed_index_scan.go

```diff
@@ -295,6 +295,13 @@ func getCostedIndexScan(ctx *sql.Context, statsProv sql.StatsProvider, rt sql.Ta
 		bestStat = stats.UpdateCounts(bestStat)
 	}
 
+	if bestStat.FuncDeps().HasMax1Row() && !ctx.QProps.JoinIsSet() && !ctx.QProps.SubqueryIsSet() && len(lookup.Ranges) == 1 {
+		// Strict index lookup without a join or subquery scope will return
+		// at most one row. We could also use some sort of scope counting
+		// to check for single scope.
+		ctx.QProps.Set(sql.QPropMax1Row)
```

Comment:
Kind of concerned about the lifecycle of these properties introducing correctness issues over time.

Do you have a good sense about e.g.

* contexts are always created fresh for every new query (even in multiquery mode)
* subcontexts don't inherit them?